lib/Command/Room/Remove.php

```diff
@@ -0,0 +1,94 @@
+<?php
+declare(strict_types=1);
+/**
+ * @copyright Copyright (c) 2020 Daniel Rudolf <nextcloud.com@daniel-rudolf.de>
+ *
+ * @author Daniel Rudolf <nextcloud.com@daniel-rudolf.de>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\Talk\Command\Room;
+
+use Exception;
+use OC\Core\Command\Base;
+use OCA\Talk\Exceptions\RoomNotFoundException;
+use OCA\Talk\Manager;
+use OCP\IConfig;
+use OCP\IUserManager;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class Remove extends Base {
+	use TRoomCommand;
```

Comment:
Instead of a trait adding all the functions in all the commands, just add them to the commands

Reply:
I was feeling like having it all in one place is a better solution - some of the methods are used multiple times (leading to duplicate code), some are not. Usually I'd create a service, but I don't think it's sensible to create a separate service for a command.

Most of the code in the trait shouldn't exist at all. I don't know the reason why the manager just returns `false` on errors instead of throwing exceptions with proper messages, or why it is an error to change something to its current value. So the best solution IMO would be to refactor the manager accordingly. However, I'm definitely not enough into the code base yet to estimate the work this requires - and it's kinda out of the scope of this PR.

So there's IMO no "perfect" solution right now. I'll choose whatever solution you prefer (trait, duplicate functions, separate service, refactor manager, …) - just let me know :+1:

Reply:
Ah, the Create conversation uses them too.

And yeah the manager just grew and was never organized.